handler/openid/validator.go

```diff
@@ -102,6 +102,9 @@ func (v *OpenIDConnectRequestValidator) ValidatePrompt(req fosite.AuthorizeReque
 
 	claims := session.IDTokenClaims()
 	if claims.Subject == "" {
+		if stringslice.Has(prompt, "none") {
+			return errors.WithStack(fosite.ErrLoginRequired.WithDebug("Cannot authenticate user silently because user's session has expired or it has never been authenticated."))
+		}
 		return errors.WithStack(fosite.ErrServerError.WithDebug("Failed to validate OpenID Connect request because session subject is empty."))
```

Comment:
Thank you! This is actually a bit different here. If `claims.Subject` is empty it implies that the fosite framework is used incorrectly. `claims.Subject` has to be hydrated because otherwise the ID token is invalid. That is the point of this check.

I think there is an error `fosite.ErrMisconfiguration` which would make this more obvious than `fosite.ErrServerError`, so from my view it would be ok to change the error.